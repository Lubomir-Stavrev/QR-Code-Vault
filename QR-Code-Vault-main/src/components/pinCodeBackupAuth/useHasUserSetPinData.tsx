import {useQuery} from 'react-query';
import {
  hasUserSetPinCode,
  deleteUserPinCode,
} from '@haskkor/react-native-pincode';

const pinCodeKeychainName = 'pincode';

export const useHasUserSetPinData = () => {
  const {isLoading, isError, isSuccess, data} = useQuery(
    'hasUserSetPinData',
    () => hasUserSetPinCode(pinCodeKeychainName),
  );

  return {isLoading, isError, isSuccess, data};
};
