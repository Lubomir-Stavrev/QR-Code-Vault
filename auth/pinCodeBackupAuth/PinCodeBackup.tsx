import React, { FC } from 'react';
import {
    StyleSheet,
    Text,
    View,
    TouchableOpacity,
} from 'react-native';

import * as Keychain from 'react-native-keychain';
import PINCode, { hasUserSetPinCode, deleteUserPinCode } from '@haskkor/react-native-pincode'


const pinCodeKeychainName = 'pincode'
const defaultPasswordLength = 6;
interface Props {
    onSuccesfullAuthentication: () => void;
}
const PinCodeBackup: FC<Props> = (props) => {


    function isPinCodeSetted() {
        return hasUserSetPinCode(pinCodeKeychainName)
            .then(res => res);
    }

    const changePin = async () => {
        let result = await deleteUserPinCode(pinCodeKeychainName);
        console.log(result);

    }

    const savePinInKeyChain = async (pin) => {

        try {
            let credentials = await Keychain.setInternetCredentials(pinCodeKeychainName, pinCodeKeychainName, pin);
            console.log(credentials);
        }
        catch (error) {
            //handle error
            console.log('Keychain couldn\'t be accessed!', error);
        }

    }

    return (
        <>
            {isPinCodeSetted() ?
                <>
                    <PINCode
                        callbackErrorTouchId={(e) => console.log(e)}
                        alphabetCharsVisible={false}
                        iconButtonDeleteDisabled={true}
                        stylePinCodeDeleteButtonText={{ marginTop: "30%", fontSize: 20, fontWeight: "900" }}
                        stylePinCodeTextButtonCircle={{ fontSize: 40, fontWeight: '300' }}
                        stylePinCodeCircle={{ width: 20, height: 5 }}
                        colorCircleButtons={'rgba(75,75,75,1)'}
                        passwordLength={defaultPasswordLength}
                        stylePinCodeTextTitle={styles.pinContainerStyle}
                        styleLockScreenTitle={styles.pinContainerStyle}
                        status={'enter'}
                        finishProcess={() => props.onSuccesfullAuthentication()}
                    />
                    <View style={{ position: 'absolute', bottom: "10%", alignSelf: 'center' }}>
                        <TouchableOpacity onPress={() => changePin()}>
                            <Text style={{
                                fontSize: 25,
                                color: '#202020',
                                fontWeight: "300",

                            }}>
                                change pin
                            </Text>
                        </TouchableOpacity>
                    </View>
                </>
                : <PINCode
                    alphabetCharsVisible={false}
                    iconButtonDeleteDisabled={true}
                    stylePinCodeDeleteButtonText={{ marginTop: "30%", fontSize: 20, fontWeight: "900" }}
                    stylePinCodeTextButtonCircle={{ fontSize: 40, fontWeight: '300' }}
                    stylePinCodeCircle={{ width: 20, height: 5 }}
                    colorCircleButtons={'rgba(70,70,70,1)'}
                    passwordLength={defaultPasswordLength}
                    stylePinCodeTextTitle={styles.pinContainerStyle}
                    styleLockScreenTitle={styles.pinContainerStyle}
                    status={'choose'}
                    stylePinCodeTextSubtitle={{ fontSize: 20, fontWeight: "400" }}
                    finishProcess={(pin) => savePinInKeyChain(pin)}
                />}
        </>
    );
};

const styles = StyleSheet.create({
    pinContainerStyle: {
        fontSize: 30,
        fontWeight: "600"
    },
    sectionTitle: {
        fontSize: 24,
        fontWeight: '600',
    },
    sectionDescription: {
        marginTop: 8,
        fontSize: 18,
        fontWeight: '400',
    },
    highlight: {
        fontWeight: '700',
    },
    authContainer: {
        flex: 1,
        height: '100%',
        alignItems: 'center',
        backgroundColor: '#202020'
    }
});

export default PinCodeBackup;
