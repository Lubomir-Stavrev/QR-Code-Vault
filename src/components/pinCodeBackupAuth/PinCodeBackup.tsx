import React, {FC, useState} from 'react';
import {ActivityIndicator} from 'react-native';
import styles from '../../styles/AuthStyles';

import * as Keychain from 'react-native-keychain';
import PINCode, {hasUserSetPinCode} from '@haskkor/react-native-pincode';
import Snackbar from 'react-native-snackbar';
import {useQuery} from 'react-query';

const pinCodeKeychainName = 'pincode';
const defaultPasswordLength = 6;

interface Props {
  onSuccesfullAuthentication: () => void;
}

const PinCodeBackup: FC<Props> = props => {
  const [isPinCodeSettedByUser, setIsPinCodeSettedByUser] = useState<
    boolean | null
  >(false);
  const [hasPinCodeValidationFailed, setHasPinCodeValidationFailed] = useState<
    boolean | null
  >(false);
  const [hasPinCodeSignUpFailed, setHasPinCodeSignUpFailed] = useState<
    boolean | null
  >(false);
  const [errorMessage, setErrorMessage] = useState<string | null>();

  const {isLoading, isError} = useQuery('hasUserSetPinData', () =>
    hasUserSetPinCode(pinCodeKeychainName).then(result =>
      setIsPinCodeSettedByUser(result),
    ),
  );

  if (isError) {
    setIsPinCodeSettedByUser(false);
    setErrorMessage('Pin code validation failed');
    setHasPinCodeValidationFailed(true);
  }

  const savePinInKeyChain = (pin: string | undefined) => {
    if (pin) {
      Keychain.setInternetCredentials(
        pinCodeKeychainName,
        pinCodeKeychainName,
        pin,
      ).then(() => setIsPinCodeSettedByUser(true));
    }
  };

  const handleFailedSignUp = () => {
    setHasPinCodeSignUpFailed(true);
    setErrorMessage('Pin code SignUp failed.');
  };

  return (
    <>
      {isLoading ? (
        <ActivityIndicator size="large" />
      ) : isPinCodeSettedByUser ? (
        <>
          <PINCode
            callbackErrorTouchId={e => console.log(e)}
            alphabetCharsVisible={false}
            iconButtonDeleteDisabled={true}
            stylePinCodeDeleteButtonText={styles.pinText}
            stylePinCodeTextButtonCircle={styles.pinText}
            colorCircleButtons={'rgba(75,75,75,1)'}
            passwordLength={defaultPasswordLength}
            stylePinCodeTextTitle={styles.pinText}
            styleLockScreenTitle={styles.pinText}
            status={'enter'}
            finishProcess={() => props.onSuccesfullAuthentication()}
          />
        </>
      ) : (
        <>
          <PINCode
            alphabetCharsVisible={false}
            iconButtonDeleteDisabled={true}
            stylePinCodeDeleteButtonText={styles.pinText}
            stylePinCodeTextButtonCircle={styles.pinText}
            colorCircleButtons={'rgba(70,70,70,1)'}
            passwordLength={defaultPasswordLength}
            stylePinCodeTextTitle={styles.pinText}
            styleLockScreenTitle={styles.pinText}
            status={'choose'}
            stylePinCodeTextSubtitle={styles.pinText}
            onFail={() => handleFailedSignUp()}
            finishProcess={pin => savePinInKeyChain(pin)}
          />
          {hasPinCodeValidationFailed || hasPinCodeSignUpFailed
            ? Snackbar.show({
                text: errorMessage ?? 'Something went wrong',
                duration: Snackbar.LENGTH_LONG,
              })
            : null}
        </>
      )}
    </>
  );
};

export default PinCodeBackup;
