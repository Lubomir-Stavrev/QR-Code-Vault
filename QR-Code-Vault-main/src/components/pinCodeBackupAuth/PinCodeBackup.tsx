import React, {FC} from 'react';
import {ActivityIndicator} from 'react-native';
import styles from '../../styles/AuthStyles';

import PINCode from '@haskkor/react-native-pincode';
import Snackbar from 'react-native-snackbar';

import {useHasUserSetPinData} from './useHasUserSetPinData';
import {useSavePinInKeyChain} from './useSavePinInKeyChain';

const defaultPasswordLength = 6;

interface Props {
  navigation: {navigate: (text: string) => void};
}

const PinCodeBackup: FC<Props> = props => {
  const hasUserSetPinData = useHasUserSetPinData();
  const savePinInKeyChain = useSavePinInKeyChain();

  return (
    <>
      {hasUserSetPinData.isLoading ? (
        <ActivityIndicator testID="ActivityIndicator" size="large" />
      ) : hasUserSetPinData.isSuccess ? (
        <PINCode
          alphabetCharsVisible={false}
          iconButtonDeleteDisabled={true}
          stylePinCodeDeleteButtonText={styles.pinText}
          stylePinCodeTextButtonCircle={styles.pinText}
          colorCircleButtons={'rgba(75,75,75,1)'}
          passwordLength={defaultPasswordLength}
          stylePinCodeTextTitle={styles.pinText}
          styleLockScreenTitle={styles.pinText}
          status={'enter'}
          finishProcess={() => props.navigation.navigate('QRCodeMenu')}
        />
      ) : (
        <>
          <PINCode
            alphabetCharsVisible={false}
            iconButtonDeleteDisabled={true}
            stylePinCodeDeleteButtonText={styles.pinText}
            stylePinCodeTextButtonCircle={styles.pinText}
            colorCircleButtons={'rgba(70,70,70,1)'}
            passwordLength={defaultPasswordLength}
            stylePinCodeTextTitle={styles.pinText}
            styleLockScreenTitle={styles.pinText}
            status={'choose'}
            stylePinCodeTextSubtitle={styles.pinText}
            finishProcess={pin => savePinInKeyChain.mutate(pin)}
          />
          {hasUserSetPinData.isError || savePinInKeyChain.isError
            ? Snackbar.show({
                text: hasUserSetPinData.isError
                  ? 'Pin code validation failed'
                  : 'Pin code SignUp failed',
                duration: Snackbar.LENGTH_LONG,
              })
            : null}
        </>
      )}
    </>
  );
};

export default PinCodeBackup;
