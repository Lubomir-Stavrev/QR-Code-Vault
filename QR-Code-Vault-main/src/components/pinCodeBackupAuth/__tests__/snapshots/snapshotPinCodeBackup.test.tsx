import React from 'react';
import PinCodeBackup from '../../PinCodeBackup';
import {useHasUserSetPinData} from '../../useHasUserSetPinData';
import renderer from 'react-test-renderer';

jest.mock('@haskkor/react-native-pincode', () => jest.fn());
jest.mock('react-query', () => ({
  useQueryClient: jest.fn(),
  useMutation: jest.fn(),
}));
jest.mock('react-native-snackbar', () => ({
  show: jest.fn(),
}));

jest.mock('../../useHasUserSetPinData', () => ({
  useHasUserSetPinData: jest.fn(),
}));
jest.mock('../../useSavePinInKeyChain', () => ({
  useSavePinInKeyChain: jest.fn(),
}));
interface UserSetPinData {
  isLoading?: boolean;
  isError?: boolean;
  isSuccess?: boolean;
  data?: any;
}
const mockUseHasUserSetPinData =
  useHasUserSetPinData as jest.Mock<UserSetPinData>;
const navigation = {navigate: jest.fn()};
describe('PinCodeBackup', () => {
  it('renders correctly', async () => {
    mockUseHasUserSetPinData.mockImplementation(() => ({isLoading: true}));
    const tree = renderer
      .create(<PinCodeBackup navigation={navigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
