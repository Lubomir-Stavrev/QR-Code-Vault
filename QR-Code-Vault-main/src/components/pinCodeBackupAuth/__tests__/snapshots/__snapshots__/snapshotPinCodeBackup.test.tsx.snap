// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinCodeBackup renders correctly 1`] = `
<ActivityIndicator
  size="large"
  testID="ActivityIndicator"
/>
`;
